File: server/cmd/server/structure/schedule/schedule.go
package schedule

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"server/cmd/server/structure/channel"
)

type ScheduleData struct {
	ScheduleId string `bson:"scheduleId" json:"id"`
	Title string `json:"title"`
	Type string `json:"type"`
	PublishedAt string  `bson:"publishedAt" json:"published_at"`
	AvailableAt string `bson:"availableAt" json:"available_at"`
	StartScheduled string `bson:"startScheduled" json:"start_scheduled"`
	Duration int `json:"duration"`
	Status string `json:"status"`
	Channel channel.ChannelData `json:"channel"`
 }

type ResponseScheduleData struct {
	ScheduleId string `bson:"scheduleId" json:"id"`
	Title string `json:"title"`
	Type string `json:"type"`
	PublishedAt primitive.DateTime  `bson:"publishedAt" json:"published_at"`
	AvailableAt primitive.DateTime `bson:"availableAt" json:"available_at"`
	StartScheduled primitive.DateTime `bson:"startScheduled" json:"start_scheduled"`
	Duration int `json:"duration"`
	Status string `json:"status"`
	Channel channel.ChannelData `json:"channel"`
}

File: mobile/lib/models/schedule.dart
import 'package:holovn_mobile/models/channel.dart';
import 'package:json_annotation/json_annotation.dart';
part 'schedule.g.dart';

@JsonSerializable(explicitToJson: true)
class Schedule{
  final String id;
  final String title;
  final String type;
  final DateTime publishedAt;
  final DateTime availableAt;
  final DateTime startScheduled;
  final int duration;
  final String status;
  final Channel channel;

  Schedule(this.id, this.title, this.type, this.publishedAt, this.availableAt, this.startScheduled, this.duration, this.status, this.channel);
  factory Schedule.fromJson(Map<String, dynamic> json) => _$ScheduleFromJson(json);
  Map<String, dynamic> toJson() => _$ScheduleToJson(this);
}